lib/Doctrine/Migrations/Finder/Finder.php

```diff
@@ -39,35 +42,52 @@ protected function getRealPath(string $directory) : string
      *
      * @return string[]
      */
-    protected function loadMigrations(array $files, ?string $namespace) : array
+    protected function loadMigrations(array $files) : array
     {
-        $migrations = [];
-
-        uasort($files, $this->getFileSortCallback());
-
+        $includedFiles = [];
         foreach ($files as $file) {
             static::requireOnce($file);
-            $className = basename($file, '.php');
-            $version   = (string) substr($className, 7);
+            $includedFiles[] = realpath($file);
+        }
 
+        $classes  = $this->loadMigrationClasses($includedFiles);
+        $versions = [];
+        foreach ($classes as $class) {
+            $version = substr($class->getShortName(), 7);
             if ($version === '0') {
                 throw new InvalidArgumentException(sprintf(
                     'Cannot load a migrations with the name "%s" because it is a reserved number by doctrine migrations' . PHP_EOL .
                     'It\'s used to revert all migrations including the first one.',
                     $version
                 ));
             }
-
-            $migrations[$version] = sprintf('%s\\%s', $namespace, $className);
+            $versions[$version] = $class->getName();
         }
 
-        return $migrations;
+        ksort($versions, SORT_STRING);
+
+        return $versions;
     }
 
-    protected function getFileSortCallback() : callable
+    /**
+     * Look up all declared classes and find those classes contained
+     * in the give `$files` array.
+     *
+     * @param string[] $files The set of files that were `required`
+     * @return ReflectionClass[] the classes in `$files`
+     */
+    protected function loadMigrationClasses(array $files) : array
     {
-        return function (string $a, string $b) {
-            return (basename($a) < basename($b)) ? -1 : 1;
-        };
+        $classes = [];
+        foreach (get_declared_classes() as $class) {
+            $ref = new ReflectionClass($class);
```

Comment:
We like to avoid using abbreviations. Please use `$reflectionClass` as the variable name here.